datasets/has_part/has_part.py

```diff
@@ -0,0 +1,139 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This dataset is a new knowledge-base (KB) of hasPart relationships, extracted from a large corpus of generic statements. Complementary to other resources available, it is the first which is all three of: accurate (90% precision), salient (covers relationships a person may mention), and has high coverage of common terms (approximated as within a 10 year old’s vocabulary), as well as having several times more hasPart entries than in the popular ontologies ConceptNet and WordNet. In addition, it contains information about quantifiers, argument modifiers, and links the entities to appropriate concepts in Wikipedia and WordNet."""
+
+from __future__ import absolute_import, division, print_function
+
+import ast
+import os
+from collections import defaultdict
+
+import datasets
+
+
+_CITATION = """\
+@misc{bhakthavatsalam2020dogs,
+      title={Do Dogs have Whiskers? A New Knowledge Base of hasPart Relations},
+      author={Sumithra Bhakthavatsalam and Kyle Richardson and Niket Tandon and Peter Clark},
+      year={2020},
+      eprint={2006.07510},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+}
+"""
+
+_DESCRIPTION = """\
+This dataset is a new knowledge-base (KB) of hasPart relationships, extracted from a large corpus of generic statements. Complementary to other resources available, it is the first which is all three of: accurate (90% precision), salient (covers relationships a person may mention), and has high coverage of common terms (approximated as within a 10 year old’s vocabulary), as well as having several times more hasPart entries than in the popular ontologies ConceptNet and WordNet. In addition, it contains information about quantifiers, argument modifiers, and links the entities to appropriate concepts in Wikipedia and WordNet.
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = "https://allenai.org/data/haspartkb"
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+
+class HasPart(datasets.GeneratorBasedBuilder):
+    @property
+    def manual_download_instructions(self):
+        return """\
+  You should download the dataset from https://allenai.org/data/haspartkb
+  Download hasPartKB.tsv to a dir of your choice,
+  which will be used as a manual_dir, e.g. `~/.manual_dir/has_part`
+  The data can then be loaded via:
+  `datasets.load_dataset("has_part", data_dir="~/.manual_dir/has_part")`.
```

Comment:
Looks like the data can be downloaded without requiring manual download
Here is the url from allenai : 
- hasPartKB.tsv: https://drive.google.com/uc?export=download&id=1Ev4RqWcPsLI9rgOGAKh-_dFKqcEZ1u-G

You can use the dl_manager to download the resource as usual.